File: python/Best Time to Buy and Sell Stock II.py
'''
Q: Best Time to Buy and Sell Stock II
Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/

'''

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit = 0
        for i in range(len(prices) - 1):
            maxProfit += max(0, prices[i + 1] - prices[i])
        return maxProfit
                
                

File: java/Best Time to Buy and Sell Stock II.java
/**
* Q: Best Time to Buy and Sell Stock II
* S: Count any profit
* Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
*/

public class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int maxProfit = 0;
        
        for (int i = 0; i < prices.length; i++) {
            if (min < prices[i]) {
                maxProfit += prices[i] - min;
            }
            min = prices[i];
        }
        return maxProfit;
    }
}